### PR TITLE
[th2-3409] ordering fields

### PR DESCRIPTION
**Pull request by Dmitriy-Yugay**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/fix/client/FixMessageFactory.java [hunk 1]
```diff
@@ -87,9 +87,9 @@ public Message create(String beginString, String msgType) {
     }
 
 
-    public FixMessage create(String msgType, int[] fieldOrder) {
+    public FixMessage create(String msgType,int[] bodyFieldOrder, int[] headerFieldOrder, int[] trailerFieldOrder) {
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/fix/client/FixMessageFactory.java]:
```suggestion
    public FixMessage create(String msgType, int[] headerFieldOrder, int[] bodyFieldOrder, int[] trailerFieldOrder) {
```

File: src/main/java/com/exactpro/th2/fix/client/Main.java [hunk 2]
```diff
@@ -333,11 +333,14 @@ protected Event initialize() {
                         }
 
                         if (sessionSettings.getOrderingFields() != null && sessionSettings.getOrderingFields().equals(YES_SETTING)) {
-                            fixMessage = messageFactory.create(MessageUtils.getMessageType(strMessage), dataDictionary.getOrderedFields());
+                            String msgType = MessageUtils.getMessageType(strMessage);
+                            fixMessage = messageFactory.create(msgType, dataDictionary.getMsgTypeOrderedFields(msgType),
+                                    dataDictionary.getMsgTypeOrderedFields("HEADER"), dataDictionary.getMsgTypeOrderedFields("TRAILER"));
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/fix/client/Main.java]: > "HEADER"
"TRAILER"

Create constants for them

File: src/main/java/com/exactpro/th2/fix/client/Main.java [hunk 3]
```diff
@@ -333,11 +333,14 @@ protected Event initialize() {
...
                             fixMessage.fromString(strMessage, sessionDataDictionary, dataDictionary, true);
                         } else {
                             fixMessage = new FixMessage(strMessage, sessionDataDictionary, dataDictionary);
                         }
+                        dataDictionary.validate(fixMessage, true);
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/fix/client/Main.java:346]: > true

Why?

**Review by cordwelt — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactpro/th2/fix/client/Main.java (hunk 3 above)
Comment (Dmitriy-Yugay) [reply src/main/java/com/exactpro/th2/fix/client/Main.java:346]: to validate a header in FIX messages > 5.0 version, we need 2 dictionaries, but the validate method in qfj uses only one, so errors will occur when validating the header

**Review by cordwelt — APPROVED** [review]

**Pull request merged**